test/stdlib/CodableTests.swift

```diff
@@ -76,14 +76,23 @@ func expectRoundTripEquality<T : Codable>(of value: T, encode: (T) throws -> Dat
     expectEqual(value, decoded, "\(#file):\(lineNumber): Decoded \(T.self) <\(debugDescription(decoded))> not equal to original <\(debugDescription(value))>")
 }
 
-func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T, lineNumber: Int) where T : Equatable {
+func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T, _ expectedJSON: String? = nil, lineNumber: Int) where T : Equatable {
     let inf = "INF", negInf = "-INF", nan = "NaN"
     let encode = { (_ value: T) throws -> Data in
         let encoder = JSONEncoder()
         encoder.nonConformingFloatEncodingStrategy = .convertToString(positiveInfinity: inf,
                                                                       negativeInfinity: negInf,
                                                                       nan: nan)
-        return try encoder.encode(value)
+        if #available(macOS 10.13, iOS 11.0, watchOS 4.0, tvOS 11.0, *) {
+            encoder.outputFormatting = .sortedKeys
+        }
+        let encoded = try encoder.encode(value)
+
+        if let expectedJSON = expectedJSON {
+            let stringValue = String(decoding: encoded, as: UTF8.self)
+            expectEqual(expectedJSON, stringValue, "\(#file):\(lineNumber): Encoded \(T.self) <\(stringValue)> not equal to expected <\(expectedJSON)>")
+        }
```

Comment:
The [`expectEqual`][] function will already print a similar message.

[`expectEqual`]: <https://github.com/apple/swift/blob/5ea28a7d3aad5ab4f29d8fc4466949866fdce70a/stdlib/private/StdlibUnittest/StdlibUnittest.swift#L259-L272>

```suggestion
        if let expectedJSON = expectedJSON {
            let actualJSON = String(decoding: encoded, as: UTF8.self)
            expectEqual(expectedJSON, actualJSON)
        }
```